gpt_index/readers/weaviate.py

```diff
@@ -0,0 +1,103 @@
+"""Weaviate reader."""
+
+from typing import Any, List, Optional
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.schema import Document
+
+
+class WeaviateReader(BaseReader):
```

Comment:
would it simplify things if we made one base class for a `VectorDbReader` that all faiss, pinecone, weaviate inherited from? i'm wondering if there's enough shared logic here to do that. 

since people have been asking for vector db support, this might save us time in the future if we have to add more vector db readers

Reply:
Good q. Tbh I thought about it and the interfaces between these three are actually quite different, each one has different required args during loading and query time. But definitely something to think about as I add more abstractions!